### PR TITLE
Adding transform helper functions and runner initial attempt

### PR DESCRIPTION
**Pull request by shawn-hurley**

### CODE REVIEW

**Comment by shawn-hurley** [comment]
I believe that this is ready to review 

@jmontleon @alaypatel07

File: transform/runner.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+)
+
+type Runner struct {
+	// This is where we need to put extra info
+	// This should include generic args to be passed to each Plugin
+	// This also needs to handle the options that it will need.
+	// TODO: Figure out options that the runner will need and implement here.
+}
+
+func (r *Runner) Run(object unstructured.Unstructured, plugins []Plugin) ([]byte, bool, error) {
```
[the remark this answers is not in the document]
Comment (shawn-hurley) [reply transform/runner.go:17]: Why over a list of GVKs?

File: transform/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+	return "", false
+}
+
+func addAnnotations(addedAnnotations map[string]string) (jsonpatch.Patch, error) {
+	patchJSON := `[`
+	i := 0
+	for key, value := range addedAnnotations {
+		if i == 0 {
+			patchJSON = fmt.Sprintf(`%v
+{"op": "add", "path": "/metadata/annotations/%v", "value": "%v"}`, patchJSON, key, value)
```
Comment (alaypatel07) [inline transform/kubernetes/kubernetes.go:155]: nit: I wonder if we can collect these strings into one place and then read and reference those from there.

In the past I have seen collecting assets in a common place allows for more automation[1][2]. we are in such early stages not sure if this is applicable but just curious if you have any thoughts regarding this

1. https://github.com/openshift/cluster-etcd-operator/blob/master/bindata/etcd/pod-cm.yaml
2. automation https://github.com/openshift/cluster-etcd-operator/blob/2f6c869efbe534f3e44fc1732019488521386807/Makefile#L32

Comment (shawn-hurley) [reply transform/kubernetes/kubernetes.go:155]: I could see moving these to constants at the top of the file WDYT? 

Or do you see us needing to update this on particular builds based on something? I am not seeing that but could be missing it

Comment (alaypatel07) [reply transform/kubernetes/kubernetes.go:155]: The top of the file seems fine for now.

**Pull request merged**